examples/ex71.hpp

```diff
@@ -0,0 +1,562 @@
+// Shared implementation ex71p/ex71 for the AD integrands and the manually
+// implemented integrators
+
+#ifndef EX71_HPP
+#define EX71_HPP
+
+#include "mfem.hpp"
+#include <memory>
+#include <iostream>
+#include <fstream>
+
+namespace mfem
+{
+template<typename DType, typename MVType>
+class MyQFunctorJ
+{
+public:
+   DType operator()(const Vector &vparam, MVType &uu)
+   {
+      double pp = vparam[0];
+      double ee = vparam[1];
+      double ff = vparam[2];
+
+      DType u = uu[3];
+      DType norm2 = uu[0] * uu[0] + uu[1] * uu[1] + uu[2] * uu[2];
+
+      DType rez = pow(ee * ee + norm2, pp / 2.0) / pp - ff * u;
+      return rez;
+   }
+
+   void operator()(const Vector &vparam, MVType &uu, MVType &rr)
+   {
+      double pp = vparam[0];
+      double ee = vparam[1];
+      double ff = vparam[2];
+
+      DType norm2 = uu[0] * uu[0] + uu[1] * uu[1] + uu[2] * uu[2];
+      DType tvar = pow(ee * ee + norm2, (pp - 2.0) / 2.0);
+
+      rr[0] = tvar * uu[0];
+      rr[1] = tvar * uu[1];
+      rr[2] = tvar * uu[2];
+      rr[3] = -ff;
+   }
+};
+
+typedef ADQFunctionTJ<MyQFunctorJ, 4> pLapIntegrandTJ;
+
+template<typename DType, typename MVType>
+class MyQFunctorH
+{
+public:
+   DType operator()(const Vector &vparam, MVType &uu)
+   {
+      double pp = vparam[0];
+      double ee = vparam[1];
+      double ff = vparam[2];
+
+      DType u = uu[3];
+      DType norm2 = uu[0] * uu[0] + uu[1] * uu[1] + uu[2] * uu[2];
+
+      DType rez = pow(ee * ee + norm2, pp / 2.0) / pp - ff * u;
+      return rez;
+   }
+};
+
+typedef ADQFunctionTH<MyQFunctorH> pLapIntegrandTH;
+
+//comment the line below in order to use
+//pLapIntegrandTJ for differentiation
+//the user interface for both TH and TJ versions
+//is exactly the same
+//#define USE_ADH
```

Comment:
Make this a runtime selectable option. It can be compiled no matter what but should be selectable from the command line without changing source files. Then we can also properly add this to the tests.

Reply:
it is implemented now